### PR TITLE
Judicial system: User Control Client

### PR DESCRIPTION
**Pull request by gudjong**
# User Control Client

https://app.asana.com/0/1199153462262248/1199920099505649

## What

☝️ 

## Why

To make it easier to manage users.

## Screenshots / Gifs

image
image
image

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx [hunk 1]
```diff
@@ -26,4 +37,31 @@ describe('Header', () => {
...
+            <Header pathname={Constants.USER_LIST_ROUTE} />
+          </UserProvider>
+        </Router>
+      </MockedProvider>,
+    )
+
+    // wait for login to complete
+    await screen.findByTestId('logout-button')
+
+    const logo = getByTestId('link-to-home')
```
Comment (oddsson) [inline apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx:58]: This should be `screen.getByTestId`

File: apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx [hunk 2]
```diff
@@ -26,4 +37,31 @@ describe('Header', () => {
...
+        <Router history={history}>
+          <UserProvider>
+            <Header pathname={Constants.USER_LIST_ROUTE} />
+          </UserProvider>
+        </Router>
+      </MockedProvider>,
+    )
+
+    // wait for login to complete
+    await screen.findByTestId('logout-button')
```
Comment (oddsson) [inline apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx:56]: A better way of doing this is `screen.findByRole('button', { name: user.name })`. According to https://testing-library.com/docs/queries/about/, you should always try to use *ByRole whenever you can.

**Review by oddsson — CHANGES_REQUESTED** [review]

File: apps/judicial-system/web/src/routes/Shared/ChangeUser/ChangeUser.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  const history = useHistory()
+
+  useEffect(() => {
+    if (data && id) {
+      setUser(data.user)
+    }
+  }, [data, id, setUser])
+
+  useEffect(() => {
+    document.title = 'Breyta notenda - Réttarvörslugátt'
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Shared/ChangeUser/ChangeUser.tsx:42]:
```suggestion
    document.title = 'Breyta notanda - Réttarvörslugátt'
```

File: apps/judicial-system/web/src/routes/Shared/UserForm/UserForm.tsx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+          onBlur={(event) => validateAndSetError('email', event.target.value)}
+          required
+          hasError={emailErrorMessage !== undefined}
+          errorMessage={emailErrorMessage}
+        />
+      </Box>
+      <Box marginBottom={2}>
+        <Checkbox
+          name="active"
+          label="Virkja notendann"
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Shared/UserForm/UserForm.tsx:293]:
```suggestion
          label="Virkja notandann"
```

File: apps/judicial-system/web/src/routes/Shared/Users/Users.treat.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  background: theme.color.blue100,
+  boxShadow: `inset 0px -1px 0px ${theme.color.blue200}`,
+})
+
+export const tableRowContainer = style({
+  borderBottom: `1px solid ${theme.color.blue200}`,
+  cursor: 'pointer',
+  transition: 'all .5s ease-in-out',
+
+  selectors: {
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Shared/Users/Users.treat.ts:43]: I don't know.

File: apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx:58]: But this comes from the render function: const { getByTestId } = render(....

File: apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/web/src/shared-components/Header/Header.spec.tsx:56]: ok

**Review by oddsson — APPROVED** [review]

**Pull request merged**